modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,432 @@
+      t = []
+      threads = 1 if threads <= 0
+      begin
+        1.upto(threads) do
+          t << framework.threads.spawn("Module(#{self.refname})", false, iplst.shift) do |ip_text|
+            next if ip_text.nil?
+            a = get_ptr(ip_text)
+            records |= a if a
           end
         end
+        t.map { |x| x.join }
+
+      rescue ::Timeout::Error
+      ensure
+        t.each { |x| x.kill rescue nil }
       end
+    end
+    records
+  end
 
+  def dns_wildcard_enabled?(domain)
+    records = get_a("#{Rex::Text.rand_text_alpha(16)}.#{domain}")
+    if records.blank?
+      false
+    else
+      print_warning('dns wildcard is enable OR fake dns server')
+      true
     end
   end
 
+  def save_loot(ltype, ctype, host, data,
+                filename = nil, info = nil, service = nil)
+    return if datastore['STOP_STORE_LOOT']
+    path = store_loot(ltype, ctype, host, data, filename, info, service)
+    print_good('saved file to: ' + path)
+  end
 
+  def get_ptr(ip)
+    vprint_status("reverse ip query: #{ip}")
+    resp = dns_query(ip, nil)
+    return if resp.blank? || resp.answer.blank?
 
-  def reverselkp(iprange,nssrv)
-    print_status("Running reverse lookup against IP range #{iprange}")
-    if not nssrv.nil?
-      @res.nameserver = (nssrv)
-      @nsinuse = nssrv
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::PTR
+      records << "#{r.ptr}"
+      report_host(host: ip, name: "#{r.ptr}", info: 'ip reverse')
+      vprint_good("#{ip}: PTR: #{r.ptr} ")
     end
-    ar = Rex::Socket::RangeWalker.new(iprange)
-    tl = []
-    while (true)
-      # Spawn threads for each host
-      while (tl.length < @threadnum)
-        ip = ar.next_ip
-        break if not ip
-        tl << framework.threads.spawn("Module(#{self.refname})-#{ip}", false, ip.dup) do |tip|
-          begin
-            query = @res.query(tip)
-            raise ::Rex::ConnectionError
-            query.each_ptr do |addresstp|
-              print_status("Hostname: #{addresstp} IP address: #{tip.to_s}")
-              report_note(:host => @nsinuse.to_s,
-                :proto => 'udp',
-                :sname => 'dns',
-                :port => 53 ,
-                :type => 'dns.enum',
-                :update => :unique_data,
-                :data => "#{addresstp},#{tip},A")
-            end
-          rescue ::Interrupt
-            raise $!
-          rescue ::Rex::ConnectionError
-          rescue ::Exception => e
-            print_error("Error: #{tip}: #{e.message}")
-            elog("Error running against host #{tip}: #{e.message}\n#{e.backtrace.join("\n")}")
-          end
-        end
-      end
-      # Exit once we run out of hosts
-      if(tl.length == 0)
-        break
-      end
-      tl.first.join
-      tl.delete_if { |t| not t.alive? }
+    return if records.none?
+    records
+  end
+
+  def get_a(domain)
+    vprint_status("query dns A : #{domain}")
```

Comment:
Especially in brute force mode, these `vprint`s can get a little out of control.  There isn't currently any other granularity other than VERBOSE or not-VERBOSE (for example, DEBUG), so I'd suggest just removing the vprints that happen right before the queries are tried.